Azure/azure-cli
# [Backup] `az backup vault`: Add new parameter `--public-network-access` to support enabling public network access for the backup vault

**Related command**

`az backup vault create`

**Description**

(Tested using the recoveryservices SDK version 2022-09-30-preview, will be released with SDK version [to be confirmed]) The PR adds support for the updated recoveryservices SDK that specifies a `public-network-access` flag for vaults, impacting vault creation/editing/display. Vault display (`az backup vault show`) is handled by the existing code, we need to modify behaviour of vault create/delete (Both handled by `az backup vault create`).

**Testing Guide**

`az backup vault show`

- When querying for a vault, under "properties" we should see the "publicNetworkAccess" field.

`az backup vault create`

- With the updated SDK and without the changes, the vault cannot successfully be created: we get an "InvalidRestApiParameter" issue.
- When creating a new vault, by default the value will be "Enable". After creation we can verify this by checking the properties as specified in `az backup vault show` testing steps
- We can also set the flag manually: if we set it to "Disable", the result will be "publicNetworkAccess: Disabled", and likewise for "Enable".
- This can also be used to edit vaults within certain criteria (check: <https://learn.microsoft.com/en-us/azure/backup/private-endpoints-overview#before-you-start>).

**History Notes**


---

This checklist is used to make sure that common guidelines for a pull request are followed.
